File: app/src/main/java/social/entourage/android/api/NewsfeedRequest.java
package social.entourage.android.api;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;
import social.entourage.android.api.model.EntourageDate;
import social.entourage.android.api.model.NewsfeedItem;

/**
 * Created by mihaiionescu on 05/05/16.
 */
public interface NewsfeedRequest {

    @GET("feeds")
    Call<NewsfeedItem.NewsfeedItemWrapper> retrieveFeed(
            @Query("before") EntourageDate before,
            @Query("longitude") double longitude,
            @Query("latitude") double latitude,
            @Query("distance") int distance,
            @Query("per") int itemsPerPage,
            @Query("types") String types,
            @Query("show_my_entourages_only") boolean onlyMyEntourages,
            @Query("time_range") int timeFrame,
            @Query("show_my_partner_only") boolean onlyMyPartnerEntourages,
            @Query("announcements") String announcementsVersion,
            @Query("show_past_events") boolean showPastEvents
    );

    @GET("feeds")
    Call<NewsfeedItem.NewsfeedItemWrapper> retrieveFeedByPage(
            @Query("page") int page,
            @Query("per") int per,
            @Query("longitude") double longitude,
            @Query("latitude") double latitude,
            @Query("tour_types") String tourTypes,
            @Query("show_tours") boolean showTours,
            @Query("show_my_entourages_only") boolean onlyMyEntourages,
            @Query("entourage_types") String entourageTypes,
            @Query("time_range") int timeFrame
    );

    @GET("myfeeds")
    Call<NewsfeedItem.NewsfeedItemWrapper> retrieveMyFeeds(
            @Query("page") int page,
            @Query("per") int per,
            @Query("entourage_types") String entourageTypes,
            @Query("tour_types") String tourTypes,
            @Query("status") String status,
            @Query("created_by_me") boolean createdByMe,
            @Query("show_my_partner_only") boolean showMyPartnerOnly,
            @Query("accepted_invitation") boolean acceptedInvitation
    );

    @GET("feeds/outings")
    Call<NewsfeedItem.NewsfeedItemWrapper> retrieveOutings(
            @Query("longitude") double longitude,
            @Query("latitude") double latitude,
            @Query("starting_after") String startingAfterOutingUUID
    );
}

File: app/src/main/java/social/entourage/android/api/request/NewsfeedRequest.kt
package social.entourage.android.api.request

import com.google.gson.annotations.SerializedName
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query
import social.entourage.android.api.model.EntourageRequestDate
import social.entourage.android.api.model.feed.NewsfeedItem

/**
 * Created by mihaiionescu on 05/05/16.
 */
class NewsfeedItemResponse(@field:SerializedName("feeds") var newsfeedItems: List<NewsfeedItem>)

interface NewsfeedRequest {
    @GET("feeds")
    fun retrieveFeed(
            @Query("before") before: EntourageRequestDate?,
            @Query("longitude") longitude: Double,
            @Query("latitude") latitude: Double,
            @Query("distance") distance: Int,
            @Query("per") itemsPerPage: Int,
            @Query("types") types: String?,
            @Query("show_my_entourages_only") onlyMyEntourages: Boolean,
            @Query("time_range") timeFrame: Int,
            @Query("show_my_partner_only") onlyMyPartnerEntourages: Boolean,
            @Query("announcements") announcementsVersion: String?,
            @Query("show_past_events") showPastEvents: Boolean,
            @Query("partners_only") isPartnersOnly: Boolean
    ): Call<NewsfeedItemResponse>

    @GET("feeds")
    fun retrieveFeedByPage(
            @Query("page") page: Int,
            @Query("per") per: Int,
            @Query("longitude") longitude: Double,
            @Query("latitude") latitude: Double,
            @Query("tour_types") tourTypes: String?,
            @Query("show_tours") showTours: Boolean,
            @Query("show_my_entourages_only") onlyMyEntourages: Boolean,
            @Query("entourage_types") entourageTypes: String?,
            @Query("time_range") timeFrame: Int
    ): Call<NewsfeedItemResponse>

    @GET("myfeeds")
    fun retrieveMyFeeds(
            @Query("page") page: Int,
            @Query("per") per: Int,
            @Query("entourage_types") entourageTypes: String?,
            @Query("tour_types") tourTypes: String?,
            @Query("status") status: String?,
            @Query("created_by_me") createdByMe: Boolean,
            @Query("show_my_partner_only") showMyPartnerOnly: Boolean,
            @Query("accepted_invitation") acceptedInvitation: Boolean
    ): Call<NewsfeedItemResponse>

    @GET("feeds/outings")
    fun retrieveOutings(
            @Query("longitude") longitude: Double,
            @Query("latitude") latitude: Double,
            @Query("starting_after") startingAfterOutingUUID: String?
    ): Call<NewsfeedItemResponse>
}